06.bowling_object/lib/frame.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require_relative 'shot'
+
+class Frame
+  def initialize(idx, first_mark, second_mark = nil, third_mark = nil)
+    @idx = idx
+    @shots = [Shot.new(first_mark), Shot.new(second_mark), Shot.new(third_mark)]
+  end
+
+  def score(frames)
+    final? ? score_without_bonus : score_without_bonus + score_bonus(frames)
+  end
+
+  private
+
+  def score_without_bonus
+    @shots.sum(&:score)
+  end
+
+  def final?
+    @idx == 9
+  end
+
+  def score_bonus(frames)
+    next_frame = frames[@idx + 1]
+    if strike?
+      next_frame.score_first_shot + (next_frame.only_one_shot? ? frames[@idx + 2].score_first_shot : next_frame.score_second_shot)
```

Comment:
```suggestion
      next_frame.score_first_shot + (next_frame.only_one_shot? ? frames[@idx + 2].scores[0] : next_frame.scores[1])
```

こんなインターフェースにするとちょっとスッキリするかも？

ただ、`frame.scores`は`frame.score`の配列ではないので、命名には少し考慮がいるかもしれません。

あと、`only_one_shot?`は`strike?`に置き換えられないですかね？